rust/src/rfb/rfb.rs

```diff
@@ -227,15 +240,26 @@ impl RFBState {
                             match chosen_security_type {
                                 2 => self.state = parser::RFBGlobalState::TCVncChallenge,
                                 1 => self.state = parser::RFBGlobalState::TSClientInit,
-                                _ => return AppLayerResult::err(),
+                                _ => {
+                                    if let Some(current_transaction) = self.get_current_tx() {
+                                        RFBState::set_event(
```

Comment:
Maybe `set_event` should belong to tx and not state

Reply:
There is only ever one tx per flow. I guess I can simplify that. Thanks!